packages/grid/x-data-grid-pro/src/components/GridDetailPanel.tsx

```diff
@@ -13,30 +13,25 @@ const DetailPanel = styled(Box, {
   slot: 'DetailPanel',
   overridesResolver: (props, styles) => styles.detailPanel,
 })<{ ownerState: OwnerState }>(({ theme }) => ({
-  zIndex: 2,
   width: '100%',
```

Comment:
100% now means "as wide as the virtual scroller content" instead of "as wide as the grid".
The detail panel should have the same width as `MuiDataGrid-virtualScrollerContent`, and if its content exceeds that width - the detail panel should be scrollable horizontally:

https://codesandbox.io/p/sandbox/loving-ramanujan-2qnn87

Reply:
I've implemented the behavior you describe here, but there's one test that seems to be different than what I understand this needs to be. Can you validate if the behavior is ok? There's a `@cherniavskii` tag in the test in question.

Reply:
My previous content was wrong.
The detail panel should actually be "as wide as the virtual scroller content", not "as wide as the grid".
But it should not exceed the total column width (which was the case when I added this comment).
Sorry for the confusion!

Reply:
Makes sense, it's possible to implement the viewport-width behavior if we leave it as it is. But it wouldn't be possible to have row-width panels if we do viewport-width. It's restored.